lib/ansible/modules/cloud/amazon/cloudformation_stack_set.py

```diff
@@ -0,0 +1,664 @@
+                OperationId=operation_ids[-1],
+            )
+        else:
+            operation_ids.append('Ansible-StackInstance-Update-{0}'.format(operation_uuid))
+            cfn.update_stack_instances(
+                StackSetName=module.params['name'],
+                Accounts=list(set(acct for acct, region in existing_stack_instances)),
+                Regions=list(set(region for acct, region in existing_stack_instances)),
+                OperationPreferences=get_operation_preferences(module),
+                OperationId=operation_ids[-1],
+            )
+        for op in operation_ids:
+            await_stack_set_operation(
+                module, cfn, operation_id=op,
+                stack_set_name=module.params['name'],
+                max_wait=module.params.get('wait_timeout'),
+            )
+
+    elif state == 'absent':
+        if not existing_stack_set:
+            module.exit_json(msg='Stack set {0} does not exist'.format(module.params['name']))
+        if module.params.get('purge_stack_instances') is False:
+            pass
+        try:
+            cfn.delete_stack_set(
+                StackSetName=module.params['name'],
+            )
+            module.exit_json(msg='Stack set {0} deleted'.format(module.params['name']))
+        except is_boto3_error_code('OperationInProgressException') as e:
+            module.fail_json_aws(e, msg='Cannot delete stack {0} while there is an operation in progress'.format(module.params['name']))
+        except is_boto3_error_code('StackSetNotEmptyException'):
+            delete_instances_op = 'Ansible-StackInstance-Delete-{0}'.format(operation_uuid)
+            #AWSRetry.jittered_backoff(retries=3, delay=10, catch_extra_error_codes=['OperationInProgressException'])(
+            cfn.delete_stack_instances(
+                StackSetName=module.params['name'],
+                Accounts=module.params['accounts'],
+                Regions=module.params['regions'],
+                RetainStacks=(not module.params.get('purge_stacks')),
+                OperationId=delete_instances_op
+            )
+            await_stack_set_operation(
+                module, cfn, operation_id=delete_instances_op,
+                stack_set_name=stack_params['StackSetName'],
+                max_wait=module.params.get('wait_timeout'),
+            )
+            cfn.delete_stack_set(
+                StackSetName=module.params['name'],
+            )
+            module.exit_json(changed=True, msg='Stack set {0} deleted'.format(module.params['name']))
+
+        # absent state is different because of the way delete_stack works.
+        # problem is it it doesn't give an error if stack isn't found
+        # so must describe the stack first
+        raise NotImplementedError("Oooooops")
+
+        module.exit_json(msg='Stack set {0} deleted'.format(module.params['name']))
+
+    result.update(**describe_stack_tree(module, stack_params['StackSetName'], operation_ids=operation_ids))
```

Comment:
If wait is False this describe call will essentially always fail (triggering OperationInProgressException) since API calls are being made for state=present when nothing is changing. If comparisons are done before making the API calls that would fix part of it, but not sure if there's anything to be done for it when something is actually being done to the stack set. It makes me feel like waiting should be non-optional (but the wait_timeout should be modifiable) or there should be a docs warning that not setting wait to true will usually traceback (maybe since create_stack always waits regardless of wait: false it wouldn't be the end of the world to default it to True in general).